File: basic/src/main/java/com/yuanshenbin/basic/state/ErrorState.java
package com.yuanshenbin.basic.state;

import android.support.annotation.DrawableRes;

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 错误状态
 */

public abstract class ErrorState extends StateAbstract {

    public abstract void showError(@DrawableRes int resId, String... msg);

    public abstract void showError(@DrawableRes int resId);

    public abstract void showError(String... msg);
}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/state/ErrorState.kt
package com.yuanshenbin.basic.state

import androidx.annotation.DrawableRes

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 错误状态
 */
abstract class ErrorState : StateAbstract() {
    abstract fun showError(@DrawableRes resId: Int, vararg msg: String?)
    abstract fun showError(@DrawableRes resId: Int)
    abstract fun showError(vararg msg: String?)
}